lib/routes/routeBackbeat.js

```diff
@@ -59,19 +60,78 @@ function _getRequestPayload(req, cb) {
+function _checkMultipleBackendRequest(request, log) {
+    const { headers, bucketName, query } = request;
+    const { operation } = query;
+    let errMessage;
+    if ((operation === 'initiatempu' || operation === 'putobject') &&
+        headers['x-scal-version-id'] === undefined) {
+        errMessage = 'bad request: missing x-scal-version-id header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    if (operation === 'putpart' &&
+        headers['x-scal-part-number'] === undefined) {
+        errMessage = 'bad request: missing part-number header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    if ((operation === 'putpart' || operation === 'completempu') &&
+        headers['x-scal-upload-id'] === undefined) {
+        errMessage = 'bad request: missing upload-id header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    if (operation === 'putobject' &&
+        headers['x-scal-canonical-id'] === undefined) {
+        errMessage = 'bad request: missing x-scal-canonical-id header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    if (operation === 'putobject' &&
+        headers['content-md5'] === undefined) {
+        errMessage = 'bad request: missing content-md5 header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    if (headers['x-scal-storage-type'] === undefined) {
+        errMessage = 'bad request: missing x-scal-storage-type header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    if (headers['x-scal-storage-class'] === undefined) {
+        errMessage = 'bad request: missing x-scal-storage-class header';
+        log.error(errMessage);
+        return errors.BadRequest.customizeDescription(errMessage);
+    }
+    const location = locationConstraints[headers['x-scal-storage-class']];
+    const isValidLocation = location &&
+        location.type === headers['x-scal-storage-type'] &&
+        location.details.bucketName === bucketName;
+    if (!isValidLocation) {
+        errMessage = 'invalid request: invalid location constraint in request';
+        log.debug(errMessage, {
+            method: request.method,
+            bucketName: request.bucketName,
+            objectKey: request.objectKey,
+            resourceType: request.resourceType,
+        });
+        return errors.InvalidRequest.customizeDescription(errMessage);
+    }
+    return undefined;
 }
 
 /*
 PUT /_/backbeat/metadata/<bucket name>/<object key>
 PUT /_/backbeat/data/<bucket name>/<object key>
-PUT /_/backbeat/multiplebackendputobject/<bucket name>/<object key>
+PUT /_/backbeat/multiplebackenddata/<bucket name>/<object key>
```

Comment:
For those two PUT commands, why not having the simpler and more REST-friendly convention: if no PartNumber attribute is provided, it's a put of the whole object (since it's provided as a `x-scal-part-number` header it works by checking the header absence), otherwise it's a put on this specific part.

You can have a common handler for PUTs on `multiplebackenddata` and route simply to the proper function depending on the header presence.

Reply:
As discussed, will leave as is for now. 😸